packages/std/src/ibc_exports.rs

```diff
@@ -0,0 +1,242 @@
+#![cfg(all(feature = "stargate", target_arch = "wasm32"))]
+use std::fmt;
+use std::vec::Vec;
+
+use schemars::JsonSchema;
+use serde::Serialize;
+
+use crate::exports::make_dependencies;
+use crate::deps::DepsMut;
+use crate::ibc::{IbcChannel, IbcConnectResponse};
+use crate::memory::{consume_region, release_buffer, Region};
+use crate::results::ContractResult;
+use crate::serde::{from_slice, to_vec};
+use crate::types::Env;
+
+// TODO: replace with https://doc.rust-lang.org/std/ops/trait.Try.html once stabilized
+macro_rules! r#try_into_contract_result {
+    ($expr:expr) => {
+        match $expr {
+            Ok(val) => val,
+            Err(err) => {
+                return ContractResult::Err(err.to_string());
+            }
+        }
+    };
+    ($expr:expr,) => {
+        $crate::try_into_contract_result!($expr)
+    };
+}
+
+/// do_ibc_channel_open is designed for use with #[entry_point] to make a "C" extern
+///
+/// ibc_open_fn does the protocol version negotiation during channel handshake phase
+pub fn do_ibc_channel_open<E>(
+    ibc_open_fn: &dyn Fn(DepsMut, Env, IbcChannel) -> Result<(), E>,
```

Comment:
Two naming ideas for this argument:
- `<export name>_fn` as we do in all the existing cases, then this should be `ibc_channel_open_fn`
- Use `contract_fn` in all cases, as this is always the contract's implementation of the `do_<export name>`

Reply:
I would go for the second.
The first one was getting very long, and annoying with cut and paste.

Since no one (except you ;) ) looks at this code, I figured I'd shorten it. But I will standardize them all on `contract_fn`

Reply:
Jupp, good. One small step towards auto-generating all of this